deeprankcore/tools/transform.py

```diff
@@ -0,0 +1,199 @@
+    Returns
+    ----------
+    df_final (pd.DataFrame): Pandas DataFrame containing the selected features as columns per all data points in
+    hdf5_path files.   
+    """
+    if not isinstance(hdf5_path, list):
+        hdf5_path = [hdf5_path]
+
+    df_final = pd.DataFrame()
+
+    for fname in hdf5_path:
+        with h5py.File(fname, 'r') as f:
+
+            mol_key = list(f.keys())[0]
+            
+            if subset is not None:
+                mol_keys = [mol for mol, _ in f.items() if mol in subset]
+            else:
+                mol_keys = [mol for mol, _ in f.items()]
+            
+            # read available node features
+            available_node_features = list(f[f"{mol_key}/{Nfeat.NODE}/"].keys())
+            available_node_features = [key for key in available_node_features if key[0] != '_']  # ignore metafeatures
+            
+            # read available edge features
+            available_edge_features = list(f[f"{mol_key}/{Efeat.EDGE}/"].keys())
+            available_edge_features = [key for key in available_edge_features if key[0] != '_']  # ignore metafeatures
+
+            # read available targets
+            available_target_features = list(f[f"{mol_key}/{targets.VALUES}/"].keys())
+
+            if node_features == "all":
+                node_features = available_node_features
+            if edge_features == "all":
+                edge_features = available_edge_features
+            if target_features == "all":
+                target_features = available_target_features
+
+            if not isinstance(node_features, list):
+                node_features = [node_features]
+            if not isinstance(edge_features, list):
+                edge_features = [edge_features]
+            if not isinstance(target_features, list):
+                target_features = [target_features]
+
+            # check node features
+            for feat in node_features:
+                if feat not in available_node_features:
+                    raise ValueError(
+                        f"The node feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable node features: {available_node_features}"
+                    )
+            # check edge features
+            for feat in edge_features:
+                if feat not in available_edge_features:
+                    raise ValueError(
+                        f"The edge feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable edge features: {available_edge_features}"
+                    )
+            # check target features
+            for feat in target_features:
+                if feat not in available_target_features:
+                    raise ValueError(
+                        f"The target feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable target features: {available_target_features}"
```

Comment:
It seems to me that a large part of this is pretty much equivalent to the `_check_features` method of `GraphDataset`. Would it make sense to pull that function out of `GraphDataset` and make a generic one that can be called from different places to avoid redundancy?

Alternatively, depending on when in GraphDataset we call for transformation, it might not be necessary to do the checks again. Maybe we just make sure that transformation happens _after_ `_check_features`, and then it doesn't need to be checked again here.

Reply:
The problem here is that the histograms visualization is done before creating the GraphDataset object, after the querying process. 

We have two options here: either we keep it this way, or we integrate this to_pandas and hist part into methods of the DeeprankDataset class. Actually I think the latter makes sense, and this way we're not going to have the _check_features repetition anymore. Also the parameters that hdf5_to_pandas takes as input are all covered by DeeprankDataset class. What do you think?

Reply:
> Actually I think the latter makes sense

I agree,

> Anyway I would open an issue about this, because if we're not going to keep hdf5 files is not convenient to fix this now. If we'll keep the hdf5 format then it's just a matter of integrating the functions as methods in the above mentioned class.

Agree with this as well

Reply:
Opened in #315